system/internal/service/user.go

```diff
@@ -142,10 +179,14 @@ func (svc *user) FindOrCreate(user *types.User) (out *types.User, err error) {
 
 func (svc *user) Create(input *types.User) (out *types.User, err error) {
 	return out, svc.db.Transaction(func() error {
-		// Encrypt user password
 		if out, err = svc.user.Create(input); err != nil {
 			return err
 		}
+
+		if !svc.prm.CanCreateUser() {
+			return errors.New("not allowed to create users")
```

Comment:
This should be moved up before actually creating the user record.

Reply:
True. Mea culpa.